libraries/AP_Scripting/applets/RockBlock.lua

```diff
@@ -0,0 +1,888 @@
+-- Lua script to send a recieve very basic MAVLink telemetry over a
+-- Rockblock SBD satellite modem
+-- Requires https://github.com/stephendade/rockblock2mav at the GCS end
+
+-- Setup:
+-- This script requires 1 serial port:
+-- A "Script" to connect the RockBlock modem
+
+-- Usage:
+-- Use the MAVLink High Latency Control ("link hl on|off" in MAVProxy) to control
+-- whether to send or not (or use "force_hl_enable")
+-- Use the "debug_text" variable to view debugging statustexts at the GCS
+-- Use the "force_hl_enable" variable to force-enable high latency mode, instead of enabling from GCS
+
+-- Caveats:
+-- This will *only* send HIGH_LATENCY2 packets via the SBD modem. No heartbeats,
+-- no command acknowledgements, no statustexts, no parameters, etc
+-- A single HIGH_LATENCY2 packet will be send every 20 sec
+-- MAVLink 1 will be used, as it's slightly more efficient (50 vs 52 bytes for a HL2 message)
+-- Any incoming packets on the first mailbox check will be ignored (as these may be from a long time in the past)
+-- Only 1 command can be sent at a time from the GCS. Any subsequent commands will overwrite the previous command
+-- The param SCR_VM_I_COUNT may need to be increased in some circumstances
+
+-- Written by Stephen Dade (stephen_dade@hotmail.com)
+local port = serial:find_serial(0)
+
+if not port or baud == 0 then
+    gcs:send_text(0, "Rockblock: No Scripting Serial Port")
+    return
+end
+
+port:begin(19200)
+port:set_flow_control(0)
+
+--[[
+Lua Object for decoding and encoding MAVLink (V1 only) messages
+--]]
+local function MAVLinkProcessor()
+    -- public fields
+    local self = {
+        -- define MAVLink message id's
+        COMMAND_LONG = 76,
+        COMMAND_INT = 75,
+        HIGH_LATENCY2 = 235,
+        MISSION_ITEM_INT = 73
+    }
+
+    -- private fields
+    local _mavbuffer = ""
+    local _mavresult = {}
+    local _payload_len = 0
+    local _mavdecodestate = 0 -- 0=looking for marker, 1=getting header,2=getting payload,3=getting crc
+    PROTOCOL_MARKER_V1 = 0xFE
+    HEADER_LEN_V1 = 6
+    local _txseqid = 0
+
+    -- AUTOGEN from MAVLink generator
```

Comment:
What is this for? Is it the CRC for individual packets? Do we care about all of these? Why not store just the ones we care about

Reply:
Fixed - removed the CRC's we don't care about.